content/codespaces/getting-started-with-codespaces/getting-started-with-your-python-project-in-codespaces.md

```diff
@@ -0,0 +1,222 @@
+			"VARIANT": "3",
+			// Options
+			"INSTALL_NODE": "true",
+			"NODE_VERSION": "lts/*"
+		}
+	},
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": { 
+		"terminal.integrated.shell.linux": "/bin/bash",
+		"python.pythonPath": "/usr/local/bin/python",
+		"python.linting.enabled": true,
+		"python.linting.pylintEnabled": true,
+		"python.formatting.autopep8Path": "/usr/local/py-utils/bin/autopep8",
+		"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
+		"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
+		"python.linting.banditPath": "/usr/local/py-utils/bin/bandit",
+		"python.linting.flake8Path": "/usr/local/py-utils/bin/flake8",
+		"python.linting.mypyPath": "/usr/local/py-utils/bin/mypy",
+		"python.linting.pycodestylePath": "/usr/local/py-utils/bin/pycodestyle",
+		"python.linting.pydocstylePath": "/usr/local/py-utils/bin/pydocstyle",
+		"python.linting.pylintPath": "/usr/local/py-utils/bin/pylint"
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"ms-python.python"
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "pip3 install --user -r requirements.txt",
+
+	// Comment out connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
+	"remoteUser": "vscode"
+}
+```
+
+- **Name** - We can name our dev container anything, this is just the default
+- **Build** - Our build properties
+  - **Dockerfile** - In our build object, Dockerfile is a reference to the Dockerfile in the same folder that was the second file added to our project. This is the reference path.
+  - **Args**
+    - **Variant**: We only have one build argument here which is the node variant we want to use which is passed into our Dockerfile.
+- **Settings** - These are {% data variables.product.prodname_vscode %} settings we wish to set
+  - **Terminal.integrated.shell.linux** - While bash is the default here, we could use zsh for example by modifying this.
+- **Extensions** - These are extensions included by default.
+  - **ms-python.python** - The Microsoft Python extension provides rich support for the Python language (for all actively supported versions of the language: >=3.6), including features such as IntelliSense, linting, debugging, code navigation, code formatting, refactoring, variable explorer, test explorer, and more.
+- **forwardPorts** - By default we can forward a port, like port 3000, but these will also forward automatically
+- **postCreateCommand** - If we want to run anything after we land in our codespace that’s not defined in our Dockerfile, like `pip3 install -r requirements`, we can do that here.
+- **remoteUser** - We’re running as the vscode user, but you can optionally set this to root
```

Comment:
Would a Python user ever set this as something besides vscode or root?

Reply:
I think the [general guidance](https://code.visualstudio.com/docs/remote/containers-advanced#_creating-a-nonroot-user) is to leave as-is: 

> Running your application as a non-root user is recommended even in production (since it is more secure)

I don't recall seeing anything in our docs where we highlight a language-specific scenario for vscode versus root.